projects/kit/components/input-count/input-count.component.ts

```diff
@@ -296,12 +301,18 @@ export class TuiInputCountComponent
         }
     }
 
-    private formatNumber(value: number): string {
-        return formatNumber(
-            value,
-            null,
-            this.numberFormat.decimalSeparator,
-            this.numberFormat.thousandSeparator,
-        );
+    private formatNumber(value: number | null): string {
+        return this.isNotNumber(value)
+            ? ``
+            : formatNumber(
+                  value,
+                  null,
+                  this.numberFormat.decimalSeparator,
+                  this.numberFormat.thousandSeparator,
+              );
+    }
+
+    private isNotNumber(value: number | null): value is null {
+        return Number.isNaN(value as any) || !isPresent(value);
```

Comment:
If you change order, you won't need to typecast

Reply:
fixed